src/stactools/modis/builder.py

```diff
@@ -289,7 +289,10 @@ def asset_info(self, key: str) -> Optional[AssetInfo]:
         if asset_info is None:
             return None
 
-        if self.metadata.qa_percent_cloud_cover:
+        if (
+            self.metadata.qa_percent_cloud_cover
+            and self.metadata.qa_percent_cloud_cover.get(key, None)
```

Comment:
Just checking whether this value can be non-truthy, non-none (like `0`)?

Reply:
Looking in the debugger, it is either a populated dictionary or `None`. Coming out of the metadata class it is either a populated dictionary or an empty dictionary; I'm not seeing where the conversion to `None` is happening. Regardless, both an empty dictionary and `None` are falsy, so fine to leave as is, I think.

Reply:
All that said, I saw that I needed an explicit check for `None` in the metadata class. Added that in latest commit.